commands/audit/audit_test.go

```diff
@@ -638,3 +626,111 @@ func TestAuditWithPartialResults(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateResultsContext(t *testing.T) {
+	mockWatches := []string{"watch-1", "watch-2"}
+	mockProjectKey := "project"
+	mockArtifactoryRepoPath := "repo/path"
+
+	tests := []struct {
+		name                    string
+		xrayVersion             string
+		expectedPlatformWatches xrayApi.ResourcesWatchesBody
+	}{
+		{
+			name:                    "Git Repo Url Supported",
+			xrayVersion:             xrayServices.MinXrayVersionGitRepoKey,
+			expectedPlatformWatches: xrayApi.ResourcesWatchesBody{GitRepositoryWatches: mockWatches},
+		},
+		{
+			name:        "Git Repo Url Not Supported (Backward Compatibility)",
+			xrayVersion: "1.0.0",
+		},
+	}
+	for _, test := range tests {
+		testCaseExpectedGitRepoHttpsCloneUrl := ""
+		expectedIncludeVulnerabilitiesIfOnlyGitRepoUrlProvided := false
+		if len(test.expectedPlatformWatches.GitRepositoryWatches) > 0 {
+			// Even if providing the git repo url, the generated object should only include the value if watches are provided
```

Comment:
I don't understand the comment. 
Do you mean that we should include the value of gitRepoUrl only if a watch is assigned to this git_repository?

Reply:
changed